aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/SamplingRuleApplier.java

```diff
@@ -138,9 +190,12 @@ SamplingResult shouldSample(
       List<LinkData> parentLinks) {
     // Incrementing requests first ensures sample / borrow rate are positive.
     statistics.requests.increment();
+    boolean reservoirExpired = System.nanoTime() >= reservoirEndTimeNanos;
     SamplingResult result =
-        reservoirSampler.shouldSample(
-            parentContext, traceId, name, spanKind, attributes, parentLinks);
+        !reservoirExpired
+            ? reservoirSampler.shouldSample(
+                parentContext, traceId, name, spanKind, attributes, parentLinks)
```

Comment:
What is the purpose of checking for the reservoir being expired here? Maybe I'm misunderstanding the meaning of expired, but shouldn't the reservoirSampler itself understand whether the amount requests to sample in a given second are used up or not?

Reply:
I believe I answered my own question - expired here refers to the reservoir quota's TTL, not to the 1-second window for which the quota applies. Leaving the comment so you can confirm.

Reply:
Yup, after the TTL we switch back to fully fixed-rate sampling